### PR TITLE
Show creation date of comments

### LINKED ISSUE #447: Show  creation date of comments
A user requested to include the creation date of a comment to figure out whether content is still valid or not.

Implementation proposal: Include the date the square brackets of the comment author.

### PR DESCRIPTION
**Pull request by chrgernoe**
Fixes #447

### CODE REVIEW

File: app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt [hunk 1]
```diff
@@ -136,10 +137,11 @@ class SectorParser(private val _db: DatabaseWrapper) : JSONWebParser() {
         for (i in 0 until jsonComments.length()) {
             val jsonComment = jsonComments.getJSONObject(i)
             val user = jsonComment.getString("username")
+            val date = _parseCommentDate(jsonComment.getString("datum"))
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt:140]: There is also an "adatum" field which may contain the date of the latest change. Would this make more sense?

File: app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt [hunk 2]
```diff
@@ -280,4 +283,10 @@ class SectorParser(private val _db: DatabaseWrapper) : JSONWebParser() {
         }
         _db.updateRocks(updatedRocks)
     }
+
+    /**
+     * Parse comment date-times in the format yyyy-MM-dd HH:mm:ss
+     * and return only the date in the format yyyy-MM-dd.
+     */
+    private fun _parseCommentDate(date: String): String = DateUtils.formatDate(date.split(" ")[0])
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt:291]: - [ ] You should add error handling here in case `date` is an empty string.

**Review by paetz — DISMISSED** [review]

File: app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt (hunk 1 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt:140]: I had a look into the PHP code of [sandsteinklettern.de](sandsteinklettern.de) and for my understanding it is the archive date. It is updated, when the comment gets updated.

Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt]: I'm now prioritizing the `adatum` and use `datum` as fallback.

**Review by paetz — APPROVED** [review]

**Pull request merged**